### ISSUE TITLE
what is the motivation for v2 refactory?

### ISSUE THREAD

**Issue by clockfly** [NVIDIA/tensorrt-inference-server#1115]
I see there is pull request
`Add GRPC V2 API endpoint. #1094`
 https://github.com/NVIDIA/tensorrt-inference-server/pull/1094

Is there a design doc that can be shared for this refractory?

**Comment by deadeyegoodwin** [comment]
We have learned a lot working on TRTIS and from that we think the REST and GRPC APIs can be significantly simplified and improved (especially the REST API). We have also been working in conjunction with the KFServing community to come up with a "generic" API that could be implemented across different inference servers (with the ability to have optional extensions for non-required functionality). The V2 APIs are the outcome and we have started work on them but it will likely take a while before that are ready for use.

There is a document describing the core API here: https://docs.google.com/document/d/1IXeoRRQZ5khrZb_2-zVsKwYU7E9pAkKVgWlXarlqA78/edit?usp=sharing

TRTIS will implement this as well as several extensions (e.g. shared-memory, model-control), but those extensions are not yet defined.
